aws/resource_aws_dynamodb_global_table.go

```diff
@@ -0,0 +1,330 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDynamoDbGlobalTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDynamoDbGlobalTableCreate,
+		Read:   resourceAwsDynamoDbGlobalTableRead,
+		Update: resourceAwsDynamoDbGlobalTableUpdate,
+		Delete: resourceAwsDynamoDbGlobalTableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(1 * time.Minute),
+			Delete: schema.DefaultTimeout(1 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsDynamoDbGlobalTableName,
+			},
+
+			"replica": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceAwsDynamoDbGlobalTableReplicaHash,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDynamoDbGlobalTableCreate(d *schema.ResourceData, meta interface{}) error {
+	dynamodbconn := meta.(*AWSClient).dynamodbconn
+
+	globalTableName := d.Get("name").(string)
+
+	input := &dynamodb.CreateGlobalTableInput{
+		GlobalTableName:  aws.String(globalTableName),
+		ReplicationGroup: expandAwsDynamoDbReplicas(d.Get("replica").(*schema.Set).List()),
+	}
+
+	log.Printf("[DEBUG] Creating DynamoDB Global Table: %#v", input)
+	_, err := dynamodbconn.CreateGlobalTable(input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(globalTableName)
+
+	log.Println("[INFO] Waiting for DynamoDB Global Table to be created")
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			dynamodb.GlobalTableStatusCreating,
+			dynamodb.GlobalTableStatusDeleting,
```

Comment:
I added this as edge-case protection if we miss `ACTIVE` between check intervals for an immediate update or something like deletion happens outside Terraform during the creation. 

When `StateChangeConf` sees an unexpected state, it will complain with an error that seems less friendly than our refresh function's error: `Error on retrieving DynamoDB Global Table when waiting` or hitting a timeout.